### ISSUE TITLE
datatypes(map) siblings creates separate entries in solr. [JIRA: RIAK-1684]

### ISSUE THREAD

**Issue by mogadanez** [basho/yokozuna#423]
I have map bucket with some data( with neseted maps )
aprox structure( don't know does it matter or not ):

```
{
    "update": {
        "some_register": "value",
        "some_flag": "disable",
        "nested_map": {
            "update": {
                "nested1_map": {
                    "update": {
                                       "some_register": "value",
                                        "some_flag": "disable",
                    }
                },
                "nested1_map": {
                    "update": {
                                       "some_register": "value",
                                        "some_flag": "disable",
                    }
                }           
            }
        },
        "some_counter": 13,
    }
}
```

Updates works fine. Even simultaneous
But Sometimes  I need recreate entire value, so I Delete it using 

```
curl -XDELETE http://host:8098/types/maps/buckets/mybucket/keys/some
```

After that sometimes  siblings  with tthumbstone appears.

```
--XZ98hy0TJbr4sVETS44XBEJf7Yt
Last-Modified: Thu, 02 Oct 2014 11:29:15 GMT

E
��A�6-.... some binary
--XZ98hy0TJbr4sVETS44XBEJf7Yt
Last-Modified: Thu, 02 Oct 2014 11:29:03 GMT
X-Riak-Deleted: true


--XZ98hy0TJbr4sVETS44XBEJf7Yt--
```

Further updates to datatype using http://host:8098/types/maps/buckets/mybucket/datatypes/some NOT create new sibling. It is replace previous one, so i always have max 2 siblings, one  with record and one thumb-stone.

Problem:
- I have also search index on this bucket.
  _each_  further  updates to this datatype( before resolution), creates new records in solr because 
  _yz_id includes sibling id:
  1_maps_mybucket_some_34*46hGXxyhuW3yn3L8bRHIml
  So, in riak I have two active siblings, in solr documents multiplied. 

Good news that when I delete record again all entries in solr deleted too.

**Comment by DSomogyi** [comment]
Comment for Jira.

**Comment by mogadanez** [comment]
Actually, priority of this is grow for datatypes, because   when use  
interfaces FetchMap, UpdateMap for get/update map and DeleteValue for removing it, there is no way  to get know about this conflict.

my scenario is: 

```
1. UpdateMap,
2. DeleteValue, 
3. UpdateMap
```

conflict appears  on step 3.  but UpdateMap itself report  OK, and return single value
further calls to FetchMap returns single value too and no info about conflicts.
at same time search index return multiple values for this item( that is multiplied with each call of UpdateMap)

Now I need  extra call to FetchValue just for ensure that there is no conflicts after update my map

**Comment by mogadanez** [comment]
Also, only way to resolve it - is removing object
both 'UpdateMap' over   and  resave object via 'StoreValue'   with proper vclock  - remove siblings  in Riak, but produce another sibling in solr

**Comment by zeeshanlakhani** [comment]
@mogadanez closing this. Fixes will be in a 2.0.7 we've been working on.
